new-frontend/src/app/pages/role/read/role-read.component.ts

```diff
@@ -0,0 +1,125 @@
+import { HttpErrorResponse } from '@angular/common/http';
+import { Component, HostBinding, Input, OnInit } from '@angular/core';
+import { TranslateService } from '@ngx-translate/core';
+import { Role, RoleLazyProperties } from 'src/app/models/api/role.model';
+import { OperationType, ResourceType, Rule, ScopeType } from 'src/app/models/api/rule.model';
+import { TableData } from 'src/app/models/application/table.model';
+import { PermissionService } from 'src/app/services/permission.service';
+import { RoleService } from 'src/app/services/role.service';
+import { RuleService } from 'src/app/services/rule.service';
+
+@Component({
+  selector: 'app-role-read',
+  templateUrl: './role-read.component.html',
+  styleUrls: ['./role-read.component.scss']
+})
+export class RoleReadComponent implements OnInit {
+  @HostBinding('class') class = 'card-container';
+
+  @Input() id = '';
+
+  constructor(
+    private roleService: RoleService,
+    private ruleService: RuleService,
+    private translateService: TranslateService,
+    private permissionService: PermissionService
+  ) {}
+
+  isLoading = true;
+  isEditing: boolean = false;
+
+  canEdit = false;
+  role?: Role;
+  roleRules: Rule[] = [];
+  allRules: Rule[] = [];
+  userTable?: TableData;
+
+  /* Bound variables to permission matrix. */
+  preselectedRules: Rule[] = [];
+  selectableRules: Rule[] = [];
+  fixedSelectedRules: Rule[] = [];
+
+  changedRules?: Rule[];
+  errorMessage?: string;
+
+  async ngOnInit(): Promise<void> {
+    this.canEdit = this.permissionService.isAllowed(ScopeType.ANY, ResourceType.ROLE, OperationType.EDIT);
```

Comment:
I don't think this check will always yield the right result. E.g. if someone has permission to edit roles of their own organization, and the currently displayed role is from another organization, this goes wrong. 
Also, default roles should never be editable/deleteable.

Reply:
In that case the error is handled later on (once you select the role). We have to define a strategy how to handle authorization in general.

Reply:
In line 122 it is handled that default roles cannot be edited.